scanner/decisions/0005-scannerv4-read-cvss-data.md

```diff
@@ -0,0 +1,36 @@
+# 0005 - Adapt Scanner V4 NVD CVSS data updating pipeline to integrate with NVD CVE Api
+
+- **Author(s):** Yi Li <yli3@redhat.com>
+- **Created:** [2023-10-10]
+
+## Status
+
+Accepted
+
+## Context
+
+The National Vulnerability Database (NVD) has announced the retirement of its JSON feeds by December 15, 2023. This creates the need to change our existing CVSS data updater pipeline for Scanner V4, which currently relies on these JSON feeds. 
+
+The NVD JSON feed offers bundled data on a yearly basis, spanning from 2002 to the present year. A benefit is that only two requests are needed in the workflow for each year (one for the meta file and one for the data bundle). However, a limitation is the absence of filtering capabilities, such as by CVSS score or modification date. In contrast, the NVD API allows data retrieval through specific HTTP URL parameters, such as publish date, modified date, or CVSS score, enabling more precise data acquisition.
+
+## Decision
+
+The updated NVD CVSS GitHub Workflow (CI) has transitioned to sourcing data from the NVD CVE API, rather than the NVD Json feeds, ensuring the timely update of the NVD CVSS data bundle in Google Storage. To fetch data, the workflow downloads based on both the start index and CVSS v3 severity. For instance, a command such as curl "https://services.nvd.nist.gov/rest/json/cves/2.0?cvssV3Severity=CRITICAL&startIndex=0" > critical-1.json would be used. For every severity level, the workflow iteratively exhaust the startIndex to capture all associated data. And there are four severity levels in total. After all json files downloaded, they will be compress to one data bundle as one single zip file.
```

Comment:
Nit:

```suggestion
The updated NVD CVSS GitHub Workflow (CI) has transitioned to sourcing data from the NVD CVE API, rather than the NVD Json feeds, ensuring the timely update of the NVD CVSS data bundle in Google Storage. To fetch data, the workflow downloads based on both the start index and CVSS v3 severity. For instance, a command such as `curl "https://services.nvd.nist.gov/rest/json/cves/2.0?cvssV3Severity=CRITICAL&startIndex=0" > critical-1.json` would be used. For every severity level, the workflow iteratively exhaust the startIndex to capture all associated data. And there are four severity levels in total. After all json files downloaded, they will be compress to one data bundle as one single zip file.
```